### PR TITLE
fix for static initialization issues in physics_naming

### PR DESCRIPTION
**Pull request by tradowsk**
`make check` failed all test due to segfaults with GCC 4.8.4. @pbauman traced issue back to static initialization fiasco in Physics Naming.

Encapsulated static `_suffix` variable in static function to fix.

All tests now passing.

### CODE REVIEW

File: src/physics/include/grins/physics_naming.h [hunk 1]
```diff
@@ -60,140 +60,144 @@ namespace GRINS
...
-    { return "ConvectionDiffusion"+_suffix; }
+    { return "ConvectionDiffusion"+_suffix(); }
 
   private:
 
     static std::string physics_name_delimiter()
     { return ":"; }
 
-    static std::string _suffix;
+    static std::string& _suffix()
```
Comment (pbauman) [inline src/physics/include/grins/physics_naming.h]: Please name the function `suffix`.

File: src/physics/include/grins/physics_naming.h [hunk 2]
```diff
@@ -60,140 +60,144 @@ namespace GRINS
...
+    {
+      static std::string suffix;
```
Comment (pbauman) [inline src/physics/include/grins/physics_naming.h]: Please name this `_suffix`.

File: src/physics/include/grins/physics_naming.h [hunk 3]
```diff
@@ -37,10 +37,10 @@ namespace GRINS
   public:
 
     static void set_suffix( const std::string& suffix )
-    { _suffix = suffix; }
+    { _suffix() = suffix; }
```
Comment (pbauman) [inline src/physics/include/grins/physics_naming.h]: Let's be explicit: `PhysicsNaming::suffix()` (assuming the function name change).

**Comment by pbauman** [comment]
Thanks! Please squash in those changes, post here that you pushed (so I get the email) and then it should be good to merge.

Note I also tripped this when I tried tip of master with GCC 4.8.5. (I've been developing with GCC 5.3.0.). Looks like we'll need a few compiler versions in the CI (server is supposed to arrive at UB on 4/22).

File: src/physics/include/grins/physics_naming.h (hunk 1 above)
Comment (tradowsk) [inline src/physics/include/grins/physics_naming.h]: If I change the function name to `suffix`, then it clashes with the parameter`suffix` on line 39 (`set_suffix()`). Any preferences for what to rename that?

Comment (pbauman) [inline src/physics/include/grins/physics_naming.h]: > If I change the function name to suffix, then it clashes with the parametersuffix on line 39 

Not if you call the function as `PhysicsNaming::suffix()`

Comment (tradowsk) [inline src/physics/include/grins/physics_naming.h]: Yeah, this note wasn't up yet when I saw your other comment. Apologies

**Comment by tradowsk** [comment]
Made the requested changes, and verified `make check` still passing

**Comment by pbauman** [comment]
Congrats @tradowsk on your first `fixup` commit and thanks for the fix.

**Pull request merged**